packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -0,0 +1,1277 @@
+  /// context.
+  ///
+  /// This method is typically used by [SliverReorderableList] item widgets that
+  /// insert or remove items in response to user input.
+  ///
+  /// If no [SliverReorderableList] surrounds the context given, this function
+  /// will return null.
+  ///
+  /// See also:
+  ///
+  ///  * [of], a similar function that will throw if no [SliverReorderableList]
+  ///    ancestor is found.
+  static SliverReorderableListState? maybeOf(BuildContext context) {
+    assert(context != null);
+    return context.findAncestorStateOfType<SliverReorderableListState>();
+  }
+}
+
+/// The state for a sliver list that allows the user to interactively reorder
+/// the list items.
+///
+/// An app that needs to start a new item drag or cancel an existing one
+/// can refer to the [SliverReorderableList]'s state with a global key:
+///
+/// ```dart
+/// GlobalKey<SliverReorderableListState> listKey = GlobalKey<SliverReorderableListState>();
+/// ...
+/// SliverReorderableList(key: listKey, ...);
+/// ...
+/// listKey.currentState.cancelReorder();
+/// ```
+///
+/// [ReorderableDragStartListener] and [ReorderableDelayedDragStartListener]
+/// refer to their [SliverReorderableList] with the static
+/// [SliverReorderableList.of] method.
+class SliverReorderableListState extends State<SliverReorderableList> with TickerProviderStateMixin {
+  final Map<int, _ReorderableItemState> _items = <int, _ReorderableItemState>{};
+
+  bool _reorderingDrag = false;
+  bool _autoScrolling = false;
+  OverlayEntry? _overlayEntry;
+  _ReorderableItemState? _dragItem;
+  _DragInfo? _dragInfo;
+  int? _insertIndex;
+  Offset? _finalDropPosition;
+  MultiDragGestureRecognizer<MultiDragPointerState>? _recognizer;
+
+  late ScrollableState _scrollable;
+  Axis get _scrollDirection => axisDirectionToAxis(_scrollable.axisDirection);
+  bool get _reverse =>
+    _scrollable.axisDirection == AxisDirection.up ||
+    _scrollable.axisDirection == AxisDirection.left;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _scrollable = Scrollable.of(context)!;
+  }
+
+  @override
+  void didUpdateWidget(covariant SliverReorderableList oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.itemCount != oldWidget.itemCount) {
+      cancelReorder();
+    }
+  }
+
+  @override
+  void dispose() {
+    _dragInfo?.dispose();
+    super.dispose();
+  }
+
+  /// Initiate the dragging of the item at [index] that was started with
+  /// the pointer down [event].
+  ///
+  /// The given [recognizer] will be used to recognize and start the drag
+  /// item tracking and lead to either an item reorder, or a cancelled drag.
+  ///
+  /// Most applications will not use this directly, but will wrap the item
+  /// (or part of the item, like a drag handle) in either a
+  /// [ReorderableDragStartListener] or [ReorderableDelayedDragStartListener]
+  /// which call this for the application.
```

Comment:
call this for the application => call this method when they detect the gesture that triggers a drag start/